integration/prefetch-test.ts

```diff
@@ -15,662 +15,466 @@ import { PlaywrightFixture } from "./helpers/playwright-fixture.js";
-    let appFixture: AppFixture;
+test.describe("prefetch=intent (hover)", () => {
+  let fixture: Fixture;
+  let appFixture: AppFixture;
 
-    test.beforeAll(async () => {
-      fixture = await createFixture(fixtureFactory("render"));
-      appFixture = await createAppFixture(fixture);
-    });
+  test.beforeAll(async () => {
+    fixture = await createFixture(fixtureFactory("intent"));
+    appFixture = await createAppFixture(fixture);
+  });
 
-    test.afterAll(() => {
-      appFixture.close();
-    });
+  test.afterAll(() => {
+    appFixture.close();
+  });
 
-    test("does not render prefetch tags during SSR", async ({ page }) => {
-      let res = await fixture.requestDocument("/");
-      expect(res.status).toBe(200);
-      expect(await page.locator("#nav link").count()).toBe(0);
-    });
+  test("does not render prefetch tags during SSR", async ({ page }) => {
+    let res = await fixture.requestDocument("/");
+    expect(res.status).toBe(200);
+    expect(await page.locator("#nav link").count()).toBe(0);
+  });
 
-    test("adds prefetch tags on hydration", async ({ page }) => {
-      let app = new PlaywrightFixture(appFixture, page);
-      await app.goto("/");
-      // Both data and asset fetch for /with-loader
-      await page.waitForSelector(
-        "#nav link[rel='prefetch'][as='fetch'][href='/with-loader?_data=routes%2Fwith-loader']",
-        { state: "attached" }
-      );
-      await page.waitForSelector(
-        "#nav link[rel='modulepreload'][href^='/build/routes/with-loader-']",
-        { state: "attached" }
-      );
-      // Only asset fetch for /without-loader
-      await page.waitForSelector(
-        "#nav link[rel='modulepreload'][href^='/build/routes/without-loader-']",
-        { state: "attached" }
-      );
-
-      // Ensure no other links in the #nav element
-      expect(await page.locator("#nav link").count()).toBe(3);
-    });
+  test("does not add prefetch tags on hydration", async ({ page }) => {
+    let app = new PlaywrightFixture(appFixture, page);
+    await app.goto("/");
+    expect(await page.locator("#nav link").count()).toBe(0);
   });
 
-  test.describe("prefetch=render (fog of war)", () => {
-    let appFixture: AppFixture;
+  test("adds prefetch tags on hover", async ({ page }) => {
+    let app = new PlaywrightFixture(appFixture, page);
+    await app.goto("/");
+    await page.hover("a[href='/with-loader']");
+    await page.waitForSelector(
+      "#nav link[rel='prefetch'][as='fetch'][href='/with-loader.data']",
+      { state: "attached" }
+    );
+    // Check href prefix due to hashed filenames
+    await page.waitForSelector(
+      "#nav link[rel='modulepreload'][href^='/assets/with-loader-']",
+      { state: "attached" }
+    );
+    await page.waitForSelector(
+      "#nav link[rel='modulepreload'][href^='/assets/with-props-']",
+      { state: "attached" }
+    );
+    await page.waitForSelector(
+      "#nav link[rel='modulepreload'][href^='/assets/index-']",
```

Comment:
These two are new with the =Vite template.  I _think_ they're legit but unsure if they should be included as assets at every route or if they should be already handled by the root or something...

Best to view this diff with whitespace hidden: https://github.com/remix-run/react-router/pull/12176/files?diff=unified&w=1

Reply:
Here's the manifest in the e2 test - these common chunks are assets for every route
 - `with-props-` is the HOC used to add our typesafe props to the component
 - `index-` seems to be the chunk that includes `react`, which is why I thought adding that to `optimizeDeps` above would fix it - but it still seems to show up and thus why we need to assert for it here to get the test to pass

![Screenshot 2024-10-22 at 12 48 18 PM](https://github.com/user-attachments/assets/ca92401f-a169-4efd-9268-38ae12c64f6d)

Reply:
The screenshot was from my testing on the rollup build.  Once I rebased onto dev to the new `tsup` build the name is `chunk-...` but the behavior is the same